barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/ultra_circuit_builder.cpp

```diff
@@ -481,16 +484,19 @@ template <typename FF> void UltraCircuitBuilder_<FF>::create_ecc_dbl_gate(const
     bool can_fuse_into_previous_gate = true;
     can_fuse_into_previous_gate = can_fuse_into_previous_gate && (w_r[this->num_gates - 1] == in.x1);
     can_fuse_into_previous_gate = can_fuse_into_previous_gate && (w_o[this->num_gates - 1] == in.y1);
+    can_fuse_into_previous_gate = can_fuse_into_previous_gate && (q_arith[this->num_gates - 1] == 0);
```

Comment:
Note: we need this because q_m = 1, could imply an arithmetic gate. We cannot check if q_elliptic will always equal 0.

This implicitly assumes that q_arith cannot be set to one at the same time as q_elliptic. This assumption is valid because those are two completely separate relations